### DISCUSSION TITLE
ADC, uDSC ...

### DISCUSSION THREAD

**Discussion by malaouar** [dbuezas/lgt8fx#307] [Q&A]
@dbuezas 
1- Have you tested the ADC quality? I've read that it's noisy and the 12-bit resolution is of limited use !!!
https://wolles-elektronikkiste.de/en/lgt8f328p-lqfp32-boards
2- Can you write, PLEASE, a tutorial about all the extra features the LGT8F328P has compared to the ATMEGA328P?
   Especially the ADC and the uDSC.
Many thanks in advance.

**Comment by wollewald** [comment]
@malaouar, what I also wrote in my blog was:

_Maybe the noise can be reduced by stabilizing the supply voltage – I have to experiment a bit more._

So, this statement refers to using the ADC just as is with the Arduino functions and no stabilization of the supply voltage or the signal itself.

**Reply by dbuezas** [reply]
There's also an Aref pin than can be used as external analog reference. That's a way to isolate the ADC from supply noise.

**Comment by malaouar** [comment]
Hi wollewald, glad that you participate to this discussion.
Your LGT reviews are awesome, thanks very much.
Yes effectiveley, you have written that, I appologize for the ommission.
Have you done any new tests on the ADC?
Can you, please,  write on the uDSC 16bit accelerator?

**Comment by wollewald** [comment]
@malaouar , I need to familiarize myself first with uDSC. I had not looked at it in detail yet. And I am not so sure if the "average reader" of my blog will be able to take advantage of this feature.
To the other question: A blog about internal ADCs and how to obtain reliable results from them is still on my to dolist. This will not only cover the LGT8F328P, but also other MCUs. So, not started yet working on this.

**Comment by malaouar** [comment]
Thanks wollewad,
We are waiting for your excellent work as usual.

**Comment by pixelwaster** [comment]
Some months back I tried to figure out the acceleration stuff. Had some samples from the LGT site (no longer there). Put together a test program and could either not get it to compile or output when it did compile. Gave up after a week of poling at it. Uploaded it to GitHub for later. 

I have some notes on the IR format, the .ino, .h, and .S asm files.  The S file drove me up the wall until I realized the test files I downloaded were in C, not C++.

https://github.com/pixelwaster/uDSC-for-LGT8FX

**Comment by malaouar** [comment]
Thanks pixelwaster, great stuff, thanks for sharing.
uDSU or UDSC? what is the correct spelling?
How to access the registers? It seems that there is a fast and a slow access!!
If some one has more info PLEASE share your notes here.

**Reply by vklimk** [reply]
Have a look here:
https://arduino.ru/forum/apparatnye-voprosy/obzor-klona-megi328-lgt8f328p#comment-392688

**Reply by beaka** [reply]
I have posted my efforts at using the uDSC functions if you are still interested. https://github.com/beaka/uDSU_DSP

A work in progress but you might find it helpful.

**Comment by malaouar** [comment]
Thanks vklimk for the link, really a great thread!!
Which is the best russian to english web translator ? google, bing or Yandex?

**Reply by vklimk** [reply]
I can recommend you to use Google Chrome browser. It has built-in translation functionality, very convenient.

**Reply by malaouar** [reply]
Thanks again vklimk
The link to the russian datasheet is broken can you upload it (the datasheet) please?
Google translation of the chinese datasheet is not so good but better than no thing!
Can you explain, please, how to use the uDSC? using assembly  and/or avr-gcc.

**Reply by vklimk** [reply]
I do not recommend you to use Russian datasheet. Use English one.
I didn't use udsc, but you can see working code in the post which the link point to. It is for Arduino, but you can rewrite it in asm/c if you like.

**Reply by malaouar** [reply]
Ok, is the 12bit resolution real?
How you evaluate the ADC quality?
I'm trying to use an FT232R in bitbang mode as an ISP programmer for LGT8F328P. I can read the GUID and LOCK bits (by copying blindly some functions from the available LarduinoISP). 
I'm still learning the said code and the STK500 protocol to decipher how Flash is erased/programed.
Can you help please?

**Reply by vklimk** [reply]
ADC is definitely better then atmega's ADC. You can use oversampling to increase its resolution.
As for ISP - I use another lgt8f328p arduino-like board as programmer. It works fine. So, I have never analysed larduinoisp code, I just use it.

**Reply by malaouar** [reply]
Me too I'm using a true arduino nano as ISP programmer for LGT and using an FT232R for AVRs.
So the idea is to use the FT232R for both and free the NANO for other more interesting tasks.

**Comment by malaouar** [comment]
Great work!!
Thanks a lot
